tests/sgx_zerobase/CMakeLists.txt

```diff
@@ -7,4 +7,21 @@ if (BUILD_ENCLAVES)
   add_subdirectory(enc)
 endif ()
 
-add_enclave_test(tests/sgx_zerobase sgx_zerobase_host sgx_zerobase_enc)
+add_enclave_test(tests/sgx_zerobase_macro_only sgx_zerobase_host
+                 sgx_zerobase_macro_only_enc)
+
+# Tests 0-base enclave creation when OE_SET_ENCLAVE_SGX*() macro is not
+# used and only oesign configuration file is used to set properties.
+add_enclave_test(tests/sgx_zerobase_conf_only sgx_zerobase_host
+                 sgx_zerobase_conf_only_enc_signed)
+
+# Tests 0-base enclave creation when OE_SET_ENCLAVE_SGX*() macro disables
+# 0-base enclave creation and oesign configuration file enables 0-base enclave creation.
+add_enclave_test(
+  tests/sgx_zerobase_prop_macro_disable_conf_enable sgx_zerobase_host
+  sgx_zerobase_macro_disable_conf_enable_enc_signed)
+
+# Tests 0-base enclave creation when OE_SET_ENCLAVE_SGX*() macro enables
+# 0-base enclave creation but oesign configuration file disables 0-base enclave creation.
+add_enclave_test(tests/sgx_zerobase_macro_enable_conf_disable sgx_zerobase_host
```

Comment:
will the test be bypassed in this case given that the zero-base is not enabled? I didn't see relevant check on the host.

Reply:
> will the test be bypassed in this case given that the zero-base is not enabled? I didn't see relevant check on the host.

Just added a check in host/host.cpp to return 0 when it sees a "config_disable" and not run the 0-base specific tests.